Deprecate marketing button tokens

rezrah opened the pull request:
## Summary

Part of https://github.com/github/primer/issues/1366
Closes https://github.com/github/primer/issues/1487


## List of notable changes:

<!--
E.g.

- **added** new design token for # because #
- **deprecated**  design token for # because #
- **updated** documentation for # because #
-->

- **deprecated**  design token for "mktg.btn" because we have replacements in `@primer/brand-primitives`
- **removed** new marketing tokens from v2 property files, because they won't be used at all and can be removed immediately.


```diff
- ---color-mktg-btn-bg
+ --brand-Button-background-base 
```

```diff
- --color-mktg-btn-shadow-outline
+ --brand-Button-shadow-default
```

```diff
- --color-mktg-btn-shadow-focus
+ --brand-Button-shadow-focus
```

```diff
- --color-mktg-btn-shadow-hover
+ --brand-Button-shadow-primary-hover
```

```diff
- --color-mktg-btn-shadow-muted
+ --color-mktg-btn-shadow-hover-muted
```

## What should reviewers focus on?

- Have I completed the deprecation step correctly? And did I miss any steps? 
- Do we deprecate in a minor? Or go straight to removal with a major? 

## Steps to test:

1. Check the package output to verify tokens appear in deprecated.json (via Upkg. E.g. on [canary release](https://unpkg.com/browse/@primer/primitives@0.0.0-20221230115427/dist/deprecated/colors.json))
1. Check deprecations appear in docs
1. Check the replacement tokens cover all use-cases 

## Supporting resources (related issues, external links, etc):

- [Primer Brand](https://github.com/primer/brand)

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change
- [ ] Verify the design tokens changed in this PR are expected using the diffing results in this PR

lukasoppermann reviewed (changes requested):
I think we can't remove tokens and set them to deprecated.

Either they must be fully removed and added to removed.json or deprecated but than they must stay in the code for some time.

Since you mentioned the tokens are not used I would suggest to go straight to removed in this case.

rezrah commented:
Thanks @lukasoppermann.

> I think we can't remove tokens and set them to deprecated.
> Either they must be fully removed and added to removed.json or deprecated but than they must stay in the code for some time.

In the changed files, you'll see I've removed the `json5` files, which were v2+ tokens. But I've kept the source files for v1 tokens and only deprecated them. I didn't remove and deprecate the same set of tokens AFAIK, or did I? 🤔
